src/Microsoft.Health.Dicom.Core/Features/Store/IIndexDataStore.cs

```diff
@@ -134,6 +134,15 @@ public interface IIndexDataStore
     /// <returns>A task that with list of instance metadata with new watermark.</returns>
     Task<IEnumerable<InstanceMetadata>> BeginUpdateInstanceAsync(int partitionKey, IReadOnlyCollection<long> versions, CancellationToken cancellationToken = default);
 
+    /// <summary>
+    /// Asynchronously updates a DICOM instance NewWatermark
+    /// </summary>
+    /// <param name="partitionKey">The partition key.</param>
+    /// <param name="studyInstanceUid">StudyInstanceUID to update</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>A task that with list of instance metadata with new watermark.</returns>
+    Task<IEnumerable<InstanceMetadata>> BeginUpdateInstanceAsync(int partitionKey, string studyInstanceUid, CancellationToken cancellationToken = default);
```

Comment:
I know we use `IEnumerable<T>` here, but we should use `IAsyncEnumerable<T>` in the future as appropriate (or even `IReadOnlyCollection<T>`)

Reply:
Changed to IReadOnlyList